pkg/odo/cli/catalog/describe/component.go

```diff
@@ -212,16 +213,12 @@ func GetDevfile(devfileComponent catalog.DevfileComponentType) (parser.DevfileOb
 			return devObj, errors.Wrapf(err, "Failed to parse registry URL for devfile component: %s", devfileComponent.Name)
 		}
 		registryURL.Path = path.Join(registryURL.Path, "devfiles", devfileComponent.Name)
-		devObj, err = devfile.ParseFromURLAndValidate(registryURL.String())
+		devObj, err = devfile.ParseDevfileAndValidate(parser.ParserArgs{URL: registryURL.String()})
 		if err != nil {
 			return devObj, errors.Wrapf(err, "Failed to download devfile.yaml from OCI-based registry for devfile component: %s", devfileComponent.Name)
 		}
 	}
 
-	err = validate.ValidateDevfileData(devObj.Data)
-	if err != nil {
-		return devObj, err
-	}
```

Comment:
This call is for performing `odo` specific validation and I think it shouldn't be removed.

Reply:
There are 9 occurrences of its removal in my PR. It looks like all of them are odo specific validations. I feel I should undo all of them. WDYT?

Reply:
Yes, we need to undo all of them.

Reply:
OK. So this wasn't necessary - https://github.com/openshift/odo/pull/4629#issuecomment-820935598 😂

Reply:
I have removed the second commit so all those changes have been removed.